src/TabViewPagerPan.js

```diff
@@ -52,77 +55,112 @@ export default class TabViewPagerPan extends Component<DefaultProps, Props, void
-        this._callResponderMethod('onPanResponderRelease', null)(...args);
-      },
-      onPanResponderMove: (...args) => {
-        this._callResponderMethod('onPanResponderMove', null)(...args);
-      },
-      onPanResponderTerminate: (...args) => {
-        this._callResponderMethod('onPanResponderTerminate', null)(...args);
-      },
-      onPanResponderTerminationRequest: (...args) => {
-        return this._callResponderMethod('onPanResponderTerminationRequest', true)(...args);
-      },
-      onShouldBlockNativeResponder: (...args) => {
-        return this._callResponderMethod('onShouldBlockNativeResponder', false)(...args);
-      },
+      onMoveShouldSetPanResponder: this._canMoveScreen,
+      onMoveShouldSetPanResponderCapture: this._canMoveScreen,
+      onPanResponderGrant: this._startGesture,
+      onPanResponderMove: this._respondToGesture,
+      onPanResponderTerminate: this._finishGesture,
+      onPanResponderRelease: this._finishGesture,
+      onPanResponderTerminationRequest: () => true,
     });
   }
 
-  componentWillReceiveProps(nextProps: Props) {
-    this._setPanHandlers(nextProps);
+  componentDidUpdate() {
+    this._normalizeSwipeVelocityThreshold();
   }
 
-  _setPanHandlers = (props: Props) => {
-    if (this.props.swipeEnabled === false) {
-      this._panHandlers = null;
+  _normalizeSwipeVelocityThreshold = () => {
+    this._swipeVelocityThreshold = this.props.swipeVelocityThreshold;
+    if (Platform.OS === 'android') {
+      // on Android, velocity is way lower due to timestamp being in nanosecond
+      // normalize it to have the same velocity on both iOS and Android
+      this._swipeVelocityThreshold /= 1000000;
+    }
+  };
+
+  _isIndexInRange = (index: number) => {
+    const { routes } = this.props.navigationState;
+    return (index >= 0 && index <= routes.length - 1);
+  };
+
+  _isMovingHorzontally = (evt: GestureEvent, gestureState: GestureState) => {
+    return (
+      (Math.abs(gestureState.dx) > Math.abs(gestureState.dy * 3)) &&
+      (Math.abs(gestureState.vx) > Math.abs(gestureState.vy * 3))
+    );
+  };
+
+  _isReverseDirection = (gestureState: GestureState) => {
+    if (this._startDirection > 0) {
+      return gestureState.vx < 0;
     } else {
-      this._panHandlers = TabViewPanResponder.forHorizontal(props);
+      return gestureState.vx > 0;
+    }
+  };
+
+  _getNextIndex = (evt: GestureEvent, gestureState: GestureState) => {
+    const { index } = this.props.navigationState;
+    if (Math.abs(gestureState.dx) > this.props.swipeDistanceThreshold || Math.abs(gestureState.vx) > this._swipeVelocityThreshold) {
+      const nextIndex = index - (gestureState.dx / Math.abs(gestureState.dx));
+      if (this._isIndexInRange(nextIndex)) {
+        return nextIndex;
+      }
+    }
+    return index;
+  };
+
+  _canMoveScreen = (evt: GestureEvent, gestureState: GestureState) => {
+    const { navigationState: { routes, index } } = this.props;
```

Comment:
I forgot this, here you should add,

```js
if (this.props.swipeEnabled === false) {
  return false;
}
```

Otherwise the prop is not used anywhere.